lib/cloud/aws/policy_statements.go

```diff
@@ -262,8 +263,12 @@ type ExternalAuditStoragePolicyConfig struct {
 
 func (c *ExternalAuditStoragePolicyConfig) CheckAndSetDefaults() error {
 	if len(c.Partition) == 0 {
-		c.Partition = "aws"
-	}
+        if strings.HasPrefix(c.Region, "us-gov") {
+            c.Partition = "aws-us-gov"
+        } else {
+            c.Partition = "aws"
```

Comment:
Let's account for the China partition as well.
We should use these two methods instead:

https://github.com/gravitational/teleport/blob/4c943c671c27812ac4729b183172b38a438894f7/api/utils/aws/region.go#L31

https://github.com/gravitational/teleport/blob/4c943c671c27812ac4729b183172b38a438894f7/api/utils/aws/region.go#L26